src/Oscoin/Crypto/Address/Serialisation.hs

```diff
@@ -50,42 +51,6 @@ data DeserializeError =
       -- ^ The input bytestring wasn't a valid base32z-encoded one
     deriving (Eq, Show)
 
--- | A 'Decoder' is simply a 'StateT' transformer which carries around the
--- binary blob being decoded (and consumed) and using 'Except' at the bottom
--- of the stack, in order to be able to report precise errors in case the
--- decoding process fails.
-type Decoder a = StateT ByteString (Except DeserializeError) a
```

Comment:
Ah here it is xD, we did call this decoder previously!

Reply:
Right, I have now separated the general-purpose monad to be called a `Parser` as it simply interprets binary blobs, whereas a `Decoder` is something which _interprets_ those bytes into a domain-specific type 😁

Reply:
:smile: not sure this makes sense - parsing has to do with language/text/syntax, so if it's related to binary, then it's not parsing :thinking: - In both cases you are transforming one representation into another.

Reply:
_Handwaving failed, target lost_ -- alternative names are accepted 😁 Should we simply call it `Get` in a similar fashion of what other libraries do (i.e. `cereal` & co)? Of course we do not enforce the same strict semantic (`Get` vs `Put` etc) but it might help conveying the idea behind those utilities.

Reply:
oh, and btw, the `cereal` library uses the term "parse" in a bunch of places/functions, so I suppose my original naming was not _that_ crazy 😬 http://hackage.haskell.org/package/cereal-0.5.8.0/docs/Data-Serialize-Get.html#t:Get

Reply:
`Get` sounds good!

> oh, and btw, the cereal library uses the term "parse/parser" in a bunch of places/functions, so I suppose my original naming was not that crazy

Maybe I'm just pedantic with English :thinking: but I'm pretty sure it's a stretch of the word :smile: :

Reply:
9e45c1b0